### PR TITLE
Fix DateMinuteParameter format

### PR DESCRIPTION
**Pull request by fphilipe**
Addresses points discussed in https://github.com/spotify/luigi/pull/625#issuecomment-125922059.

It was using H as the separator between hours and minutes, which is incorrect. The ISO 8601 standard requires a colon in the extended format or nothing in the basic format. Opt for basic format since colons don't play nice with file systems.

Added some specs for the parameter since there didn't seem to be any. Went for deprecation although I'm not entirely sure whether it is needed.

### CODE REVIEW

File: luigi/parameter.py [hunk 1]
```diff
@@ -370,11 +370,12 @@ class DateHourParameter(Parameter):
     19:00.
     """
 
-    date_format = '%Y-%m-%dT%H'  # ISO 8601 is to use 'T'
+    date_format = '%Y-%m-%dT%H'
```
Comment (erikbern) [inline luigi/parameter.py]: why did you change this line?

File: luigi/parameter.py [hunk 2]
```diff
@@ -394,11 +396,23 @@ class DateMinuteParameter(DateHourParameter):
...
+    def parse(self, s):
+        try:
+            return super(DateMinuteParameter, self).parse(s)
+        except ValueError:
+            warnings.warn(
+                'Using "H" between hours and minutes is deprecated, omit it instead.',
+                DeprecationWarning,
+                stacklevel=2
+            )
+            return datetime.datetime.strptime(s, self.deprecated_date_format)
```
Comment (erikbern) [inline luigi/parameter.py]: this will throw a deprecation warning for any misspecified string – maybe makes more sense to try deprecated first

**Comment by erikbern** [comment]
LGTM

File: luigi/parameter.py (hunk 2 above)
Comment (fphilipe) [inline luigi/parameter.py]: You're right, I'll fix it.

File: luigi/parameter.py (hunk 1 above)
Comment (fphilipe) [inline luigi/parameter.py]: I didn't see the point of the comment and since I removed the one below, I thought I'd remove it here as well. ISO 8601 is being mentioned above anyways, so it doesn't add much IMO. Also, I think it's rather arbitrary to document `T` and not `YYYY-MM-DD`, for instance.

I can undo the change on this line if you prefer :blush:

**Comment by Tarrasch** [comment]
Hmm, can you rebase this change. Unfortunately I made a patch that changes same places of code in #1098

**Comment by fphilipe** [comment]
@Tarrasch Rebased.
